ChildMindInstitute/mindlogger-app-refactor
# fix(markdown): Added html sanitizer when content includes html markdown

**Requires `yarn install`**


[sanitize-html](https://github.com/apostrophecms/sanitize-html#readme) OSS package added to MindLogger


### 📝 Description

🔗 [Jira Ticket M2-5218](https://mindlogger.atlassian.net/browse/M2-5218)

This PR changes:

- The way how markdown content is preprocessed before sending over to the WebView

### 🪤 Peer Testing

Example test step:

- Edit an applet using Postman: PUT https://api-dev.cmiml.net/applets/${applet_id}.
- Assign "<script>alert('test 3')</script>"string to the `question.en` field <img width="341" alt="image" src="https://github.com/ChildMindInstitute/mindlogger-app-refactor/assets/20073193/15323dbc-36c5-46eb-a597-f64ccbdbb2a4">
- Process the request
- Go to the mobile app and perform Pull-to-Refresh operation to fetch the latest applet update.
- Go to the activity with the changes `question.en` field.

**Expected outcome:** No alert window pops up.
